llvm/lib/DebugInfo/DWARF/DWARFExpression.cpp

```diff
@@ -240,17 +240,21 @@ static void prettyPrintBaseTypeRef(DWARFUnit *U, raw_ostream &OS,
                                    ArrayRef<uint64_t> Operands,
                                    unsigned Operand) {
   assert(Operand < Operands.size() && "operand out of bounds");
-  auto Die = U->getDIEForOffset(U->getOffset() + Operands[Operand]);
-  if (Die && Die.getTag() == dwarf::DW_TAG_base_type) {
-    OS << " (";
-    if (DumpOpts.Verbose)
-      OS << format("0x%08" PRIx64 " -> ", Operands[Operand]);
-    OS << format("0x%08" PRIx64 ")", U->getOffset() + Operands[Operand]);
-    if (auto Name = dwarf::toString(Die.find(dwarf::DW_AT_name)))
-      OS << " \"" << *Name << "\"";
+  if (U) {
+    auto Die = U->getDIEForOffset(U->getOffset() + Operands[Operand]);
+    if (Die && Die.getTag() == dwarf::DW_TAG_base_type) {
+      OS << " (";
+      if (DumpOpts.Verbose)
+        OS << format("0x%08" PRIx64 " -> ", Operands[Operand]);
+      OS << format("0x%08" PRIx64 ")", U->getOffset() + Operands[Operand]);
+      if (auto Name = dwarf::toString(Die.find(dwarf::DW_AT_name)))
+        OS << " \"" << *Name << "\"";
+    } else {
+      OS << format(" <invalid base_type ref: 0x%" PRIx64 ">",
+                   Operands[Operand]);
+    }
   } else {
-    OS << format(" <invalid base_type ref: 0x%" PRIx64 ">",
-                 Operands[Operand]);
+    OS << format(" <base_type ref: 0x%" PRIx64 ">", Operands[Operand]);
```

Comment:
Moving this to the front of the function and using an early exit would simplify the patch and reduce nesting.
```cpp
  if (!U) {
    OS << format(" <base_type ref: 0x%" PRIx64 ">", Operands[Operand]);
    return;
  }
  auto Die = ...
```

Reply:
Good suggestion, changed it.